pkg/flow/internal/controller/value_cache.go

```diff
@@ -97,18 +97,24 @@ func (vc *valueCache) ClearModuleExports() {
 	vc.mut.Lock()
 	defer vc.mut.Unlock()
 
-	vc.moduleExportsChanged = true
+	vc.moduleChangedIndex++
 	vc.moduleExports = make(map[string]any)
 }
 
-// HasModuleExportsChangedSinceLastCall returns true if the module exports changed since the last call to HasModuleExportsChangedSinceLastCall.
-func (vc *valueCache) HasModuleExportsChangedSinceLastCall() bool {
-	vc.mut.Lock()
-	defer vc.mut.Unlock()
+// HasModulesChanged returns true if the module exports index is different from changeIndex.
+func (vc *valueCache) HasModulesChanged(changeIndex int) bool {
```

Comment:
I'm not sure this function is necessary anymore, the caller can directly 
check `cache.ExportChangeIndex() != lastIndex`

Reply:
Your correct, will remove